cpp/frontend_pyreadline.cc

```diff
@@ -235,4 +242,50 @@ Readline* MaybeGetReadline() {
 #endif
 }
 
+static void readline_cb(char* line) {
+  if (line == nullptr) {
+    gReadline->latest_line_ = nullptr;
+  } else {
+    gReadline->latest_line_ = StrFromC(line);
+    free(line);
+  }
+  gReadline->ready_ = true;
+  rl_callback_handler_remove();
+}
+
+Str* readline(Str* prompt) {
+  fd_set fds;
+  FD_ZERO(&fds);
+  rl_callback_handler_install(prompt->data(), readline_cb);
+
+  gReadline->latest_line_ = nullptr;
+  gReadline->ready_ = false;
+  while (!gReadline->ready_) {
+    FD_SET(fileno(rl_instream), &fds);
+    int ec = select(FD_SETSIZE, &fds, NULL, NULL, NULL);
+    if (ec == -1) {
+      if (errno == EINTR && pyos::LastSignal() == SIGINT) {
```

Comment:
What's the difference between `SigintCount() > 0` and `pyos::LastSignal() == SIGINT` ?

Reply:
Oh whoops these should all be `SigintCount() > 0` (it's more robust to multiple signals arriving in quick succession). Nice catch!